include/highfive/h5easy_bits/H5Easy_Eigen.hpp

```diff
@@ -19,165 +19,105 @@ namespace H5Easy {
+        if (std::decay<T>::type::RowsAtCompileTime == 1) {
+            return {static_cast<size_t>(data.cols())};
+        }
+        if (std::decay<T>::type::ColsAtCompileTime == 1) {
+            return {static_cast<size_t>(data.rows())};
+        }
+        return {static_cast<size_t>(data.rows()), static_cast<size_t>(data.cols())};
     }
-    return {static_cast<size_t>(data.rows()),
-            static_cast<size_t>(data.cols())};
-}
 
+    using EigenIndex = Eigen::DenseIndex;
 
-using EigenIndex = Eigen::DenseIndex;
+    // get the shape of a "DataSet" as size 2 "std::vector<Eigen::Index>"
+    inline static std::vector<EigenIndex> shape(const File& file,
+                                                const std::string& path,
+                                                const DataSet& dataset,
+                                                int RowsAtCompileTime) {
+        std::vector<size_t> dims = dataset.getDimensions();
 
-// get the shape of a "DataSet" as size 2 "std::vector<Eigen::Index>"
-inline std::vector<EigenIndex> shape(const File& file,
-                                     const std::string& path,
-                                     const DataSet& dataset,
-                                     int RowsAtCompileTime) {
-    std::vector<size_t> dims = dataset.getDimensions();
+        if (dims.size() == 1 && RowsAtCompileTime == 1) {
+            return std::vector<EigenIndex>{1u, static_cast<EigenIndex>(dims[0])};
+        }
+        if (dims.size() == 1) {
+            return std::vector<EigenIndex>{static_cast<EigenIndex>(dims[0]), 1u};
+        }
+        if (dims.size() == 2) {
+            return std::vector<EigenIndex>{static_cast<EigenIndex>(dims[0]),
+                                           static_cast<EigenIndex>(dims[1])};
+        }
 
-    if (dims.size() == 1 && RowsAtCompileTime == 1) {
-        return std::vector<EigenIndex>{1u, static_cast<EigenIndex>(dims[0])};
+        throw detail::error(file, path, "H5Easy::load: Inconsistent rank");
     }
-    if (dims.size() == 1) {
-        return std::vector<EigenIndex>{static_cast<EigenIndex>(dims[0]), 1u};
+
+    // write to open DataSet of the correct size
+    // (use Eigen::Ref to convert to RowMajor; no action if no conversion is needed)
+    inline static void write(DataSet& dataset, const T& data) {
+        Eigen::Ref<const Eigen::Array<typename std::decay<T>::type::Scalar,
+                                      std::decay<T>::type::RowsAtCompileTime,
+                                      std::decay<T>::type::ColsAtCompileTime,
+                                      std::decay<T>::type::ColsAtCompileTime == 1
+                                          ? Eigen::ColMajor
+                                          : Eigen::RowMajor,
+                                      std::decay<T>::type::MaxRowsAtCompileTime,
+                                      std::decay<T>::type::MaxColsAtCompileTime>,
```

Comment:
I would keep one argument per line by adding a line-break before the first argument and changing the indentation (see original version)